src/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from 'react';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+
+import { InputGroup, Button } from '../../index';
+import AutoCompleteInput from './AutoCompleteInput';
+import AutoCompleteItems from './AutoCompleteItems';
+import { KEY_CODES } from '../../common/helpers';
+
+export const getActiveItems = items =>
+  items
+    .filter(
+      ({ disabled, type }) => !disabled && !['header', 'divider'].includes(type)
+    )
+    .map(({ text }) => text);
+
+class AutoComplete extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: ''
+    };
+  }
+
+  render() {
+    const {
+      onSearch,
+      labelText,
+      onInputUpdate,
+      items,
+      actionText,
+      ...rest
+    } = this.props;
+
+    const activeItems = getActiveItems(items);
+
+    return (
+      <Downshift
+        onStateChange={({ inputValue, ...rest }) => {
+          if (typeof inputValue === 'string') {
+            onInputUpdate(inputValue);
+            this.setState({ inputValue });
+          }
+        }}
+        defaultHighlightedIndex={0}
+        selectedItem={this.state.inputValue}
+        {...rest}
+      >
+        {({
```

Comment:
It just looks really weird to pass a method as children in react JSX.
Downshift API support a render prop that you can be used instead, I think it would feel better.

Reply:
Yeah I'm totally open on this concept. It's just a preference since Downshift supports both. I'll change it and we'll take a look!

Reply:
Having a render prop sounds interesting, as it would allow us to customize the options (e.g. think you would like to add an avatar in the user dropdown)

Reply:
@ohadlevy yeah that's exactly the idea here. This implementation is pushing for a standardized auto-complete using the `MenuItem` component with just text but Downshift can be used to compose any kind of dropdown interface you want.

Reply:
yep... render prop vs children as a function is just semantics i believe... either is fine w/ me 👍 ... i see people posting polls about this lol